### PR TITLE
dev: lockfile patch for use_lock process

### PR DESCRIPTION
**Pull request by cloudguruab**
This adds a `remove_lockfiles` and `check_process` method to the `use_locks` decorator to handle responses to waiting for lockfile releases within the conversion hosts. https://issues.redhat.com/secure/RapidBoard.jspa?rapidView=16007&view=detail&selectedIssue=OSP-13078&quickFilter=101301

### CODE REVIEW

**Comment by cloudguruab** [comment]
- [ ] add ci patch to separate PR

**Review by jistr — CHANGES_REQUESTED** [review]

**Comment by cloudguruab** [comment]
I agree this approach is cleaner, simply handling check/cleanup lockfiles when composing on `126` allows us to utilize check_process against the cleanup_and_check method. Keeps it DRY as well.

File: os_migrate/plugins/module_utils/workload_common.py [hunk 1]
```diff
@@ -61,6 +94,8 @@ def wait_for_lock(self):
                     self.log.debug('Released lock: %s', result)
                 except subprocess.CalledProcessError as err:
                     self.log.error('Error releasing lock: %s', str(err))
+                    check_and_cleanup_lockfiles()
```
Comment (jistr) [inline os_migrate/plugins/module_utils/workload_common.py:97]: The lock which is being created by this use_lock decorator is cleaned up on line 93 above. It probably won't hurt but if line 93 didn't succeed, i'm not sure if it's worth trying again.

**Review by jistr — CHANGES_REQUESTED** [review]
Two small things but other than that i think it's good to go!

**Review by jistr — CHANGES_REQUESTED** [review]

File: os_migrate/plugins/module_utils/workload_common.py [hunk 2]
```diff
@@ -28,18 +28,48 @@
...
+    """
+    # Remove specific lockfiles
+    for lockfile in [ATTACH_LOCK_FILE_SOURCE, ATTACH_LOCK_FILE_DESTINATION, PORT_LOCK_FILE]:
+        with open(lockfile, "r", encoding='utf-8') as f:
+            pid = int(f.read().strip())
+        if check_process(pid):
+            # print(f"Lockfile {lockfile} is currently in use by process: {pid}")
+            pass
+        else:
+            os.remove(lockfile)
```
[the remark this answers is not in the document]
Comment (cloudguruab) [reply os_migrate/plugins/module_utils/workload_common.py]: I'm testing this, but I think running the check_and_cleanup method only when an error occurs would be better as a routine step.

[the remark this answers is not in the document]
Comment (cloudguruab) [reply os_migrate/plugins/module_utils/workload_common.py]: Was able to clear this up, looks like its searching against other lockfiles in each migration step where the decorator is used. Maybe adding if conditions around the loop could work, but might cause missed failures

**Pull request merged**